### PR TITLE
chore(data-waehouse): Added job cancellation checking into rest source

### PR DESCRIPTION
**Pull request by Gilbert09**
## Problem
- If an external data job gets canceled due to the account being over the monthly sync limit, then the rest source currently will just continue to run until the end of the job

## Changes
- Early exit from source paging when the job has been canceled

## Does this work well for both Cloud and self-hosted?
Yes

## How did you test this code?
Local testing

### CODE REVIEW

File: posthog/temporal/data_imports/pipelines/helpers.py [hunk 1]
```diff
@@ -5,19 +5,13 @@
...
+async def is_job_cancelled(
     team_id: int,
     job_id: str,
-    new_count: int,
-):
+) -> bool:
     model = await aget_external_data_job(team_id, job_id)
 
-    if new_count >= CHUNK_SIZE:
-        new_count = 0
```
Comment (Gilbert09) [inline posthog/temporal/data_imports/pipelines/helpers.py:16]: This wasn't really doing anything before from what I can tell - so best to just remove it

File: posthog/temporal/data_imports/pipelines/rest_source/__init__.py [hunk 2]
```diff
@@ -244,14 +253,17 @@ def paginate_resource(
                 hooks: Optional[dict[str, Any]],
                 client: RESTClient = client,
                 incremental_object: Optional[Incremental[Any]] = incremental_object,
-                incremental_param: IncrementalParam = incremental_param,
-            ) -> Generator[Any, None, None]:
-                if incremental_object:
+                incremental_param: IncrementalParam | None = incremental_param,
+            ) -> AsyncGenerator[Iterator[Any], Any]:
+                if not is_job_cancelled(team_id=team_id, job_id=job_id):
```
Comment (EDsCODE) [inline posthog/temporal/data_imports/pipelines/rest_source/__init__.py:258]: It's aborting if the job isn't canceled?

File: posthog/temporal/data_imports/pipelines/rest_source/__init__.py [hunk 3]
```diff
@@ -291,9 +303,12 @@ def paginate_dependent_resource(
                 resolved_param: ResolvedParam = resolved_param,
                 include_from_parent: list[str] = include_from_parent,
                 incremental_object: Optional[Incremental[Any]] = incremental_object,
-                incremental_param: IncrementalParam = incremental_param,
-            ) -> Generator[Any, None, None]:
-                if incremental_object:
+                incremental_param: IncrementalParam | None = incremental_param,
+            ) -> AsyncGenerator[Any, Any]:
+                if not is_job_cancelled(team_id=team_id, job_id=job_id):
```
Comment (EDsCODE) [inline posthog/temporal/data_imports/pipelines/rest_source/__init__.py:308]: same here

File: posthog/temporal/data_imports/pipelines/rest_source/__init__.py (hunk 2 above)
Comment (Gilbert09) [reply posthog/temporal/data_imports/pipelines/rest_source/__init__.py:258]: Yeah, turns out I was very tired when I was writing this. I also don't even `await` it either 🤡

**Review by EDsCODE — APPROVED** [review]

**Pull request merged**